### PR TITLE
add test for async migrations not being required on fresh installs

### LINKED ISSUE #8306: Async migrations new installs is_required=false unittest
For new installs all is_required checks should result in 'false'. To avoid failures we can write a unittest for this, that just runs all migrations is_required checks.

### PR DESCRIPTION
**Pull request by yakkomajuri**
## Changes

*Please describe.*  
*If this affects the frontend, include screenshots.*  

Async migrations are from going from an old schema to a new one. New installs should always have the updated schema.
 
This tests people are writing good `is_required` checks. Could maybe make it more selective in terms of when it runs
Closes https://github.com/PostHog/posthog/issues/8306

## How did you test this code?

*Please describe.*

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -32,6 +32,7 @@ per-file-ignores =
     ./posthog/management/commands/partition.py: T001
     ./posthog/management/commands/sync_feature_flags.py: T001
     ./posthog/management/commands/test_migrations_are_null.py: T001
+    ./posthog/management/commands/test_async_migrations_not_required.py: T001
```
Comment (yakkomajuri) [inline .flake8:35]:
```suggestion
```

**Comment by tiina303** [comment]
Test failure is related https://github.com/PostHog/posthog/runs/4981227122?check_suite_focus=true

> Could maybe make it more selective in terms of when it runs

that's fine, can re-visit later if needed.

Could you manually test that it fails as expected, i.e. change some migration required to return True and see it fails. You probably did already, so mention in description ;)

File: conftest.py [hunk 2]
```diff
@@ -91,6 +92,14 @@ def reset_clickhouse_tables():
         sync_execute(item)
 
 
+def run_additional_setup_ops():
```
Comment (tiina303) [inline conftest.py:95]: aren't async migrations ee only?

File: ee/conftest.py [hunk 3]
```diff
@@ -91,6 +92,14 @@ def reset_clickhouse_tables():
         sync_execute(item)
 
 
+def run_additional_setup_ops():
```
Comment (tiina303) [inline ee/conftest.py:95]: why do we need to do this, why doesn't the fresh install initial migration add that comment?

Comment (yakkomajuri) [reply ee/conftest.py:95]: Because on the test environment we're not actually running the migrations - annoyingly we're setting up the DB independently

Comment (yakkomajuri) [reply ee/conftest.py:95]: `conftest` is just a `pytest` thing, you'll see this file runs all the table creation statements and truncates tables in between

File: conftest.py (hunk 2 above)
Comment (yakkomajuri) [reply conftest.py:95]: Well this ee stuff is changing and the answer is actually that going forward, no, they won't be ee-only. But anyway, the CH setup happens in both places so good to add for consistency

File: ee/conftest.py (hunk 3 above)
Comment (tiina303) [reply ee/conftest.py:95]: Oh the test just got way less useful :(

Comment (yakkomajuri) [reply ee/conftest.py:95]: it did

**Pull request merged**